LibVT: Quote the file paths on a DropEvent

mustafaquraish opened the pull request:
Earlier, we would directly just paste the file path into the terminal,
however if the path had and spaces or special characters, they would
not be escaped properly. This patch just quotes the paths before
putting them on screen.

alimpfard commented:
I don't think LibVT is the right place for this.

The shell should handle turning the path to a single argument, ideally in a context-aware way.
How about turning drop event inputs to paste actions (so bracketed paste would work), having LibLine allow customising the paste handler, and finally making the shell properly quote or escape the inserted path?

mustafaquraish commented:
> I don't think LibVT is the right place for this.
> 
> The shell should handle turning the path to a single argument, ideally in a context-aware way.
> How about turning drop event inputs to paste actions (so bracketed paste would work), having LibLine allow customising the paste handler, and finally making the shell properly quote or escape the inserted path?

Truth be told I just found the place the drag-and-drop was being handled and implemented the fix there. I like the idea of turning it into a paste and handling them the same way, but isn't it more complicated when pasting? Drag-and-drop is easier to handle since we handle each file being dragged separately.

For instance, if we just have "a b.txt" on the clipboard, it's possible that it's referring to a single file called "a b.txt" (with a space), or two separate files "a" and "b.txt" and the user wanted the space specifically. If I now have 3 files "a", "b.txt" and "a b.txt" it's not clear what a paste action should do here. Do you have any thoughts on how to go about this?

I don't know much about how LibVT works, so I'm not sure I'd even be comfortable doing this properly. Just saw a bug and fixed it 🤷‍♂️

alimpfard commented:
For the many files thing, you can just emit as many pastes.
My problem with quoting like this is that it's not at all privy to what the application receiving the path considers a quote, and that this specific implementation doesn't even consider files with `"` in their name.

In such cases, it's always best to let the receiving application decide how it wants to handle the path itself
